### PR TITLE
add package metadata files as recommended by PyPA

### PR DESCRIPTION
**Pull request by tanabarr**
additional package files added
Travis deploy to pypi on git tagging
more detailed metadata

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,16 +5,35 @@
 
 
 from setuptools import setup, find_packages
+# from distutils.core import setup, find_packages
```
Comment (jgrund) [inline setup.py]: Can we remove this?

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,11 @@
+A Python package that contains common components for the IML project
+====================================================================
+
+Different areas of the IML project utilise common code that is shared
+distributed through this package.
+
+This packaging intends to improve code reuse and componentization
+within the IML project.
+
+The main IML public repository is located at
```
Comment (jgrund) [inline README.rst]: Can we remove this line please, this will hold less true over time.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,11 @@
+A Python package that contains common components for the IML project
```
Comment (jgrund) [inline README.rst]: Why do we need a .rst file? Is this a Pypi specific thing?

File: setup.py [hunk 4]
```diff
@@ -5,16 +5,35 @@
...
-    author_email = 'tom.nabarro@intel.com',
-    url = 'https://github.com/intel-hpdd/iml-common',
-    description = 'Common library used by both agent and manager',
+    name='iml-common',
+    use_scm_version=True,
+    setup_requires=['setuptools_scm'],
+    license='MIT',
+    packages=find_packages(exclude=excludes),
+    include_package_data=True,
+    author='Intel(R) Corporation',
```
Comment (jgrund) [inline setup.py]: We've been using "IML Team" here.

File: setup.py [hunk 5]
```diff
@@ -5,16 +5,35 @@
...
+    author_email='tom.nabarro@intel.com',
```
Comment (jgrund) [inline setup.py]: Is this field required?

File: setup.py [hunk 6]
```diff
@@ -5,16 +5,35 @@
...
+    url='https://pypi.python.org/pypi/iml-common',
+    description='Common library used by multiple IML component',
```
Comment (jgrund) [inline setup.py]: component -> components

File: README.rst (hunk 3 above)
Comment (tanabarr) [inline README.rst]: yes if we don't have a README.txt or .rst we get a warning

**Pull request merged**